src/v/cluster/archival/archival_policy.cc

```diff
@@ -409,7 +409,7 @@ static ss::future<candidate_creation_result> create_upload_candidate(
       begin_inclusive, end_inclusive, segment, result, io_priority);
     if (file_range_result) {
         vlog(
-          archival_log.error,
```

Comment:
https://redpandadata.slack.com/archives/C078LGKDTJ7/p1725567006575289 logs for a test run where this happens